src/R/Components/Test/Plots/PlotIntegrationTest.cs

```diff
@@ -603,25 +1057,56 @@ public class PlotIntegrationTest : IAsyncLifetime {
             }
         }
 
+        private async Task WaitForDeviceCreatedAsync(Func<Task> action) {
+            var eas1 = new EventTaskSource<IRPlotManager>((o, h) => o.DeviceCreateMessageReceived += h, (o, h) => o.DeviceCreateMessageReceived -= h);
+            var eas2 = new EventTaskSource<IRPlotManager>((o, h) => o.ActiveDeviceChanged += h, (o, h) => o.ActiveDeviceChanged -= h);
+            var task1 = eas1.Create(_workflow.Plots);
+            var task2 = eas2.Create(_workflow.Plots);
+            await action();
+            await task1;
+            await task2;
+        }
+
+        private async Task WaitForDeviceChangedAsync(Func<Task> action) {
+            var eas = new EventTaskSource<IRPlotManager>((o, h) => o.ActiveDeviceChanged += h, (o, h) => o.ActiveDeviceChanged -= h);
+            var task = eas.Create(_workflow.Plots);
+            await action();
+            await task;
+        }
+
+        private async Task WaitForPlotAsync(IRPlotDeviceViewModel vm, Func<Task> action) {
+            var eas = new EventTaskSource<IRPlotDeviceViewModel>((o, h) => o.PlotChanged += h, (o, h) => o.PlotChanged -= h);
+            var task = eas.Create(vm);
+            await action();
+            await task;
+        }
+
         private async Task WaitForPlotAsync(Func<Task> action) {
-            var eas = new EventTaskSource<IRPlotManager>((o, h) => o.PlotChanged += h, (o, h) => o.PlotChanged -= h);
-            var plotChangedTask = eas.Create(_workflow.Plots);
+            var eas = new EventTaskSource<IRPlotManager>((o, h) => o.PlotMessageReceived += h, (o, h) => o.PlotMessageReceived -= h);
+            var task = eas.Create(_workflow.Plots);
+            await action();
+            await task;
+        }
+
+        private async Task WaitForMutated(Func<Task> action) {
+            var eas = new EventTaskSource<IRSession, EventArgs>((o, h) => o.Mutated += h, (o, h) => o.Mutated -= h);
+            var task = eas.Create(_workflow.RSession);
             await action();
-            await plotChangedTask;
+            await task;
         }
 
-        private async Task WaitForLocatorModeChangedAsync(Action action) {
-            var eas = new EventTaskSource<IRPlotManager>((o, h) => o.LocatorModeChanged += h, (o, h) => o.LocatorModeChanged -= h);
-            var locatorModeChangedTask = eas.Create(_workflow.Plots);
+        private async Task WaitForLocatorModeChangedAsync(IRPlotDeviceViewModel vm, Action action) {
+            var eas = new EventTaskSource<IRPlotDeviceViewModel>((o, h) => o.LocatorModeChanged += h, (o, h) => o.LocatorModeChanged -= h);
```

Comment:
Same thing here - let's add a static nested class to `Microsoft.R.Components.Test.EventTaskSources`